book_source/developers_guide/Github-issues.Rmd

```diff
@@ -0,0 +1,121 @@
+4. submit the bug report
+5. If you do find the answer, explain the resolution (in the issue) and close the issue
+
+### Required content
+
+Note: 
+
+* **a bug is only a bug if it is reproducible**
+* **clear bug reports save time**
+
+1.  Clear, specific title
+2.  Description - 
+ * What you did
+ * What you expected to happen
+ * What actually happened
+ * What does work, under what conditions does it fail?
+ * Reproduction steps - minimum steps required to reproduce the bug
+3. additional materials that could help identify the cause:
+ * screen shots
+ * stack traces, logs, scripts, output
+ * specific code and data / settings / configuration files required to reproduce the bug
+ * environment (operating system, browser, hardware)
+
+Requesting a feature
+--------------------
+
+(from The Pragmatic Programmer, available as
+[ebook](http://proquestcombo.safaribooksonline.com/0-201-61622-X/223)
+through UI libraries, hardcopy on David’s bookshelf)\
+
+* focus on “user stories”, e.g. specific use cases
+* Be as specific as possible, 
+
+* Here is an example:
+
+ 1.  Bob is at www.mysite.edu/maps
+ 2.  map of the the region (based on user location, e.g. US, Asia, etc)
+ 3.  option to “use current location” is provided, if clicked, map zooms in to, e.g. state or county level
+    4.  for site run:
+        1.  option to select existing site or specify point by lat/lon
+        2.  option to specify a bounding box and grid resolution in
+            either lat/lon or polar stereographic.
+
+    5.  asked to specify start and end times in terms of year, month,
+        day, hour, minute. Time is recorded in UTC not local time, this
+        should be indicated
+
+Closing an issue
+----------------
+
+1. Definition of “Done”
+ * test
+ * documentation
+2.  when issue is resolved:
+ * status is changed to “resolved”
+ * assignee is changed to original author
+3. if original author agrees that issue has been resolved
+ * original author changes status to “closed”
+4.  except for trivial issues, issues are only closed by the author 
+
+When to submit an issue?
+-----------------------
+
+### Ideally, non-trivial code changes will be linked to an issue and a commit.
+
+This requires creating issues for each task, making small commits, and referencing the issue within your commit message. Issues can be created [on GitHub](https://github.com/PecanProject/pecan/issues/new). These issues can be linked to commits by adding text such as `fixes gh-5` (see [[Using Git During Development|Using-Git#during-development]]).
+
+Rationale: This workflow is a small upfront investment that reduces error and time spent re-creating and debugging errors. Associating issues and commits, makes it easier to identify why a change was made, and potential bugs that could arise when the code is changed. In addition, knowing which issue you are working on clarifies the scope and objectives of your current task. 
+
+Label Categories
+----------------
+
+<table border="1">
```

Comment:
We should probably replace this with a rendered table from BETY, but for the moment, it would be good to replace this XML with a more human-readable markdown table.